src/wp-includes/fonts.php

```diff
@@ -122,6 +122,36 @@ function wp_get_font_dir() {
 		'error'   => false,
 	);
 
+	$made_dir = wp_mkdir_p( $defaults['path'] );
```

Comment:
I've come to realize that attempting to create the directory here is problematic.

As the code runs before the `font_dir` filter, it's unknown  at this point in the execution whether the directory path will be overridden by a plugin. If it is overridden then this code may create a junk directory in the sites filesystem that will never be used. 

What I think the code will need to do is:

* check if the folder exists and is writable
* if the folder doesn't exist, check the parent exists and is writable
* rely on `wp_upload_dir()` to create the directory after the `upload_dir` filter runs (which will include firing the `font_dir` filter) 
* check the error status produced by `wp_upload_dir()`

Reply:
> create the directory after the upload_dir filter runs (which will include firing the font_dir filter)

+1. 

Seems the first two checks:
- check if the folder exists and is writable
- if not, check the parent exists and is writable
are [also done in `wp_mkdir_p()`](https://github.com/WordPress/wordpress-develop/blob/6.4/src/wp-includes/functions.php#L2034-L2099). Somewhat unsure if they need to be done (repeated) before running wp_mkdir_p()?

Reply:
Yes, good points @peterwilsoncc. I agree 👍

Reply:
Good call; in [this commit](https://github.com/WordPress/wordpress-develop/pull/6259/commits/90a3982636f1e47209e19a8a170ce027fb921af0) I implemented the change to create the folder only after the `font_dir` filter runs.